externalpolicy/property_list.go

```diff
@@ -86,13 +151,89 @@ func (self *PropertyList) Add_Property(new_ele *Property) error {
 	return nil
 }
 
+// Validate will return an error if any property in the list has an invalid format or a value that does not match a declared type
+// if a property has no type field, the defualt used is string
 func (self *PropertyList) Validate() error {
-
-	// Check for incorrect boolean expression.
-
-	// Check for incorrect number value.
-
-	// This validation logic might be able to be borrowed from somewhere else in anax.
-
+	for _, property := range *self {
+		if property.Name == "" || property.Value == nil {
+			return fmt.Errorf("Property must include a name and a value: %v", property)
+		}
+		declaredType := property.Type
+		if declaredType == "" {
+			declaredType = STRING_TYPE
+		}
+		switch actualType := property.Value.(type) {
+		case bool:
```

Comment:
I would like the validation to be smarter. It should be ok to specify a boolean property value without the type field. When properties are compared the (runtime) type of the value has to be checked to ensure they are the same.